src/components/atoms/CodeEditor.tsx

```diff
@@ -151,6 +173,53 @@ class CodeEditor extends React.Component<CodeEditorProps> {
     window.removeEventListener('codemirror-mode-load', this.reloadMode)
   }
 
+  getExtraKeysKeyMap = (keyMapStyle: string) => {
+    return {
+      Enter: 'newlineAndIndentContinueMarkdownList',
+      Tab: 'indentMore',
+      [osName === 'macos' ? 'Cmd-F' : 'Ctrl-F']: (cm: CodeMirror.Editor) =>
+        this.handleOnLocalSearchToggle(cm, true),
+      // todo: [komediruzecki-14/02/2021] Ctrl-R used for page reload so Ctrl+H used for
+      //  local replace dialog
+      // [osName === 'macos' ? 'Cmd-R' : 'Ctrl-R']: (cm: CodeMirror.Editor) =>
+      //   this.handleOnLocalSearchReplaceToggle(cm, true),
+      'Ctrl-H': (cm: CodeMirror.Editor) =>
+        this.handleOnLocalSearchReplaceToggle(cm, true),
+      Esc: (cm: CodeMirror.Editor) => {
+        if (keyMapStyle === 'vim') {
+          return CodeMirror.Pass
+        } else {
+          this.handleOnLocalSearchToggle(cm, false)
+          return this.handleOnLocalSearchReplaceToggle(cm, false)
+        }
+      },
+    }
+  }
+
+  handleOnLocalSearchReplaceToggle = (
+    editor: CodeMirror.Editor,
+    nextState: boolean
+  ) => {
+    const { onLocalSearchReplaceToggle } = this.props
+    if (onLocalSearchReplaceToggle == null) {
+      return
+    }
+
+    onLocalSearchReplaceToggle(editor, nextState)
+  }
+
+  handleOnLocalSearchToggle = (
+    editor: CodeMirror.Editor,
+    nextState: boolean
```

Comment:
It's actually just the next boolean state if it should be shown or not.
I'll just rename it for now, if more states will be needed this can be refactored to an object with this and other properties (highly unlikely)